functions/controllers/user.js

```diff
@@ -26,51 +38,83 @@ exports.signupUser = (req, res) => {
   //validating user
   let token, userId;
   //ensuring user does not exist in db.. using unique username
-  db.doc(`/users/${newUser.username}`)
-    .get()
-    .then(doc => {
-      if (doc.exists) {
+  try {
+    const docu = await db.doc(`/users/${newUser.userName}`).get()
+      if (docu.exists) {
         return res
           .status(400)
```

Comment:
Make use of http-status-code as stipulated in your Engineering convention: https://www.npmjs.com/package/http-status-codes
See engineering convention here: https://github.com/enyeInc/Engineering-Conventions#api-endpoint